README.md

```diff
@@ -106,19 +106,128 @@ server {
+  Several additional actions will be done in this new deployment:
+  
+    1. ModSecurity and its dependencies will be installed
+    2. Default configuration for ModSecurity will be enabled
+
+- You can test that the CRS are active with the following request:
+    
+    `curl -X INVALID_HTTP_METHOD https://$YOUR_APP_NAME.osc-fr1.scalingo.io -v`
+    
+    You should expect a 403 forbidden answer such as the following:
+    
+    ```bash
+    > INVALID_HTTP_METHOD / HTTP/2
+    > Host: $YOUR_APP_NAME.osc-fr1.scalingo.io
+    > User-Agent: curl/7.64.1
+    > Accept: */*
+    >
+    * Connection state changed (MAX_CONCURRENT_STREAMS == 128)!
+    < HTTP/2 403
+    < date: Tue, 31 May 2022 13:58:46 GMT
+    < content-type: text/html; charset=utf-8
+    < content-length: 146
+    < x-request-id: 343e6a24-640e-499a-9dfe-f5dbb636ef45
+    < strict-transport-security: max-age=31536000
+    <
+    <html>
+    <head><title>403 Forbidden</title></head>
+    <body>
+    <center><h1>403 Forbidden</h1></center>
+    <hr><center>nginx</center>
+    </body>
+    </html>
+    ```
+    
+
+### Updating the CRS rules
+
+- You have to redeploy the application, the latest stable version is downloaded during the build phase. For that, create an empty commit on your repository and push it to the scalingo remote
+
+Note: you can manually set a specific version of the CRS by setting the variable `MODSECURITY_CORE_RULE_SET_VERSION` (default is `3.3.2` at the day of May 31, 2022)
+
+### Updating the ModSecurity version
+
+- Upon each deployment, the latest packaged version of modsecurity is used. Scalingo does not provide any guarantee in term of packagin time after each release, get in touch with the support if you need a specific version.
+
+Note: minimal supported version is 3.0.6
+
+### How-to add a custom rule
+
+- Note that, on Scalingo, the root of your repository is deployed on `/app`
+- Create a file to hold all the custom rules you will write and reference it in the nginx config file like so:
+
+```bash
+##############################################
+# in nginx.conf.erb file
+# This file is written in Nginx config language
+
+location / {
+    modsecurity on; # Enable ModSecurity on /
+    modsecurity_rules_file /app/custom-rules.modsecurity; # load custom rules file
+    # (...)
+    # The rest of your NGINX config file
+}
 
+##############################################
+# in custom-rules.modsecurity file:
+# This file is written in ModSecurity config language
+
+# CUSTOM RULE id:1234
+# IF query or body parameter contains a parameter named “param1” which contains “test”
+# THEN block the request with a code 403 and log the event
+SecRule ARGS:param1 "@contains test" \
+	"id:1234,\
+	 deny,\
+	 log,\
+	 status:403,\
+	 severity: 'CRITICAL',\
+	 tag: 'custom-rule',\
+	 msg: 'this is the log message you will see',\
+	 logdata: '%{MATCHED_VAR_NAME}=%{MATCHED_VAR}'"
```

Comment:
I don't understand well this part. Is it two different files to modify? If yes, I would create two code blocks.

Moreover having to create a nginx.conf.erb seems odd, shouldn't we make the ModSecurity rules file configurable via the environment with a default value set to `/app/vendor/nginx/conf/modsec-nginx.conf`?

Reply:
Definitely, two different blocks, and it should NOT be of type `bash`.

Actually it's how it work for now @EtienneM the goal of the PR is to document what exists, we could make PR for improvement, today this way is the way to go.

This way doesn't replace ` /app/vendor/nginx/conf/modsec-nginx.conf` It adds an additional rule, the default Core Rule Set is still set and enabled.